File: lab1/parser/symbol.go
package parser

type Symbol struct {
	Name  string
	Value string
}

var dictionarySymbs = []rune{'(', ')', '*', '|'}
var DictionaryNames = []string{"LP", "RP", "STAR", "OR"}

func IdxInDictionary(str rune) int {
	for idx, element := range dictionarySymbs {
		if str == element {
			return idx
		}
	}
	return -1
}

File: lab1/parser/symbol.py
dictionarySymbs = ['(', ')', '*', '|']
dictionaryDescs = ['LB', 'RB', 'STAR', 'OR']

class Symbol:
    def __init__(self, desc, value):
        self.desc = desc
        self.value = value

    def __str__(self):
        return self.name + ":" + self.value

def IdxInDictionary(char):
    for idx in range(len(dictionarySymbs)):
        if char == dictionarySymbs[idx]:
            return idx
    return -1